File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3043_find-the-length-of-the-longest-common-prefix/solutions/python3/solution.py
class Solution:
    def longestCommonPrefix(self, arr1: list[int], arr2: list[int]) -> int:
        # Convert all numbers to strings
        arr1 = [str(num) for num in arr1]
        arr2 = [str(num) for num in arr2]
        
        # Sort arrays to make use of string properties in sorted arrays
        arr1.sort()
        arr2.sort()
        
        # Helper function to find longest common prefix of two strings
        def findLCP(s1, s2):
            min_len = min(len(s1), len(s2))
            lcp_length = 0
            for i in range(min_len):
                if s1[i] == s2[i]:
                    lcp_length += 1
                else:
                    break
            return lcp_length
        
        # Compare only the first and last elements after sorting, for each array
        max_prefix_length = 0
        if arr1 and arr2:
            # Check combinations of firsts and lasts that could give longest LCP
            candidates = [
                findLCP(arr1[0], arr2[0]),
                findLCP(arr1[0], arr2[-1]),
                findLCP(arr1[-1], arr2[0]),
                findLCP(arr1[-1], arr2[-1])
            ]
            max_prefix_length = max(candidates)
        
        return max_prefix_length

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3043_find-the-length-of-the-longest-common-prefix/solutions/java/solution.java
import java.util.Arrays;

class Solution {
    public int longestCommonPrefix(int[] arr1, int[] arr2) {
        // Sort both arrays to prepare for binary search approach
        Arrays.sort(arr1);
        Arrays.sort(arr2);
        
        int maxPrefixLength = 0;
        
        for (int num1 : arr1) {
            int left = 0, right = arr2.length - 1;
            String s1 = Integer.toString(num1);
            while (left <= right) {
                int mid = left + (right - left) / 2;
                String s2 = Integer.toString(arr2[mid]);
                int commonLength = commonPrefixLength(s1, s2);
                
                maxPrefixLength = Math.max(maxPrefixLength, commonLength);
                
                // Use the nature of sorted array to decide which direction to go
                if (s2.startsWith(s1)) {
                    // No longer prefix is possible from smaller numbers in arr2
                    left = mid + 1;
                } else {
                    right = mid - 1;
                }
            }
        }
        
        return maxPrefixLength;
    }
    
    private int commonPrefixLength(String s1, String s2) {
        int minLength = Math.min(s1.length(), s2.length());
        for (int i = 0; i < minLength; i++) {
            if (s1.charAt(i) != s2.charAt(i)) {
                return i;
            }
        }
        return minLength;
    }
}
